lib/snapshot-manager.js

```diff
@@ -365,8 +386,31 @@ function determineSnapshotDir(projectDir, testDir) {
 	return testDir;
 }
 
+function determineSourceMappedDir(options) {
+	const sourceMapFilePath = path.join(options.testDir, `${options.name}.map`);
+	const sourceMapContent = tryReadSourceMap(sourceMapFilePath);
+
+	if (sourceMapContent) {
+		const targetFilename = path.basename(sourceMapContent.file, 'js');
+		const testFileSource = sourceMapContent.sources.find(file => {
+			const fileName = file.replace(path.extname(targetFilename), '');
+			return fileName.indexOf(targetFilename) > -1;
+		});
```

Comment:
Thanks I learned a few things here. I guess we wont be able to answer how much more to guard without trying more than just typescript. With the way things are layed out it shouldn't be too hard to expand on this.